src/charm.py

```diff
@@ -177,20 +180,13 @@ def _set_mimir_config(self) -> bool:
             logger.error("Failed to push updated config file: %s", e)
             raise BlockedStatusError(str(e))
 
-    def _set_alerts(self) -> bool:
-        """Create alert rule files for all Mimir consumers.
+    def _set_alerts(self) -> None:
+        """(Re-)create alert rule files for all Mimir consumers.
 
         Returns: True if alerts rules have changed, otherwise False.
         Raises: BlockedStatusError exception if PebbleError, ProtocolError, PathError exceptions
             are raised by container.remove_path
         """
-        remote_write_alerts = self.remote_write_provider.alerts()
-        alerts_hash = sha256(str(remote_write_alerts))
-        alert_rules_changed = alerts_hash != self._stored.alerts_hash
-
-        # Pushing files every time for situations such as cluster restart:
-        # Relation data and stored state match, but files haven't been written yet.
-
         try:
             self._container.remove_path(RULES_DIR, recursive=True)
         except PebbleError as e:
```

Comment:
I'm rethinking this pattern.
If a pebble operation fails after a `can_connect` guard, we probably do want the charm to go into error state, so juju retries the hooks without additional intervention.
Setting BlockedStatus is nice but we would have to wait until the next update-status for this to auto-resolve.
Wdyt?